app/components/frame.hbs

```diff
@@ -1,11 +1,13 @@
 <div>
-    <div data-test-palette>
-        Palette
-        {{#each colorOptions as option}}
-            <div color=option.name></div>
-        {{/each}}
-    </div>
-    <canvas>
-
-    </canvas>
+    {{#each this.pixelRows as |row|}}
+        <div class="flex" data-test-frame-row>
+            {{#each row as |pixel|}}
+                <Swatch
+                    data-test-pixel={{pixel.index}}
```

Comment:
Two, sorta related questions!

First, the `{{#each}}` helper will also yield an `index`, so you would not _necessarily_ to create a separate structure with an `index` in it (since you have a 2 dimensional array here, how you did it might make more sense though): 

```hbs
{{#each this.pixelRows as |row rowIndex|}}
  {{#each row as |pixel index|}}
    <Swatch data-test-pixel="({{index}},{{rowIndex}})" />
  {{/each}}
{{/each}}
```

Also, I'm curious the reason for having separate "rows" in the data structure? Is achieving this something that could be done with CSS Grid instead since this is, perhaps, a layout concern?

Reply:
Ooo, interesting. I'd created the separate structure w/ `index` to preserve a reference to the `index` from the original array of pixel colors. 
I think a possible refactor would definitely be using the built-in `{{#each}}` `index` and `rowIndex` as you suggest and doing some multiplication to find the original index.

That said, I really hadn't considered using CSS Grid - doing that would eliminate the need for rows, which would eliminate the need for the funky indexing and the separate structure to inject the index.